### PR TITLE
chore: fix broken release workflow.

### PR DESCRIPTION
**Pull request by spencerdrak**
This change fixes and issue experienced [in releasing semgrep 0.120.0](https://github.com/returntocorp/semgrep/actions/runs/3379203790) and some follow-on issues that would've popped up if that step had succeeded. 

Test plan: 

The changes in this PR were tested in go/test-gh-actions and then ported to this repo.
- [source](https://github.com/returntocorp/test-gh-actions/blob/develop/.github/workflows/release.yml)
- [successful run](https://github.com/returntocorp/test-gh-actions/actions/runs/3381581391)
- [uploaded schema file (draft release)](https://github.com/returntocorp/semgrep-interfaces/releases)

In addition, I'll be running another release soon to ensure we're good to go.

PR checklist:

- [x] Purpose of the code is [evident to future readers](https://semgrep.dev/docs/contributing/contributing-code/#explaining-code)
- [x] Tests included or PR comment includes a reproducible test plan
- [x] Documentation is up-to-date
- [x] A changelog entry was [added to changelog.d](https://semgrep.dev/docs/contributing/contributing-code/#adding-a-changelog-entry) for any user-facing change
- [x] Change has no security implications (otherwise, ping security team)

If you're unsure on any of this, please see:

- [Contribution guidelines](https://semgrep.dev/docs/contributing/contributing-code)!
- [One of the more specific guides located here](https://semgrep.dev/docs/contributing/contributing/)

### CODE REVIEW

File: .github/workflows/release.yml [hunk 1]
```diff
@@ -111,22 +113,23 @@ jobs:
...
-          path: rule_schema.yaml
+          submodules: true
+          token: ${{ steps.token.outputs.token }}
       - name: Upload Schema Files
         id: upload-semgrep-schema-files
         env:
           GITHUB_TOKEN: "${{ steps.token.outputs.token }}"
         run: |
-          gh release --repo returntocorp/semgrep-interfaces upload ${{ steps.get_version.outputs.VERSION }} rule_schema.yaml
+          gh release --repo returntocorp/semgrep-interfaces upload ${{ needs.create_release.outputs.VERSION }} cli/src/semgrep/semgrep_interfaces/rule_schema.yaml
```
Comment (aryx) [inline .github/workflows/release.yml]: I actually have a change that rename this file to rule_schema_v1.yaml, so we might need to take care when we merge it.

Comment (spencerdrak) [reply .github/workflows/release.yml]: Check, good catch. Do we want to dynamically look for and upload `rule_schema*.yaml`?

Comment (spencerdrak) [reply .github/workflows/release.yml]: I saw your PR @aryx and will rebase this off develop once that goes in.

Comment (spencerdrak) [reply .github/workflows/release.yml]: Fixed. Also removed one other unused step.

**Pull request closed without merging**